pkg/cluster/k8sres.go

```diff
@@ -490,17 +490,27 @@ func (c *Cluster) nodeAffinity(nodeReadinessLabel map[string]string, nodeAffinit
 	}
 }
 
-func generatePodAffinity(labels labels.Set, topologyKey string, nodeAffinity *v1.Affinity) *v1.Affinity {
```

Comment:
Can you make [generateLogicalBackupJob](https://github.com/zalando/postgres-operator/blob/81e17e4af1e114e2da82224cbaba323a7f4b8381/pkg/cluster/k8sres.go#L2064) also call `generatePodAffinity`, now that it can generate the preferred affinity with weight?